addons/account/models/account_move.py

```diff
@@ -3157,7 +3157,7 @@ def add_file_data_results(file_data, invoice):
                 close_file(file_data)
                 continue
 
-            decoder = self._get_edi_decoder(file_data, new=new)
+            decoder = (current_invoice or current_invoice.new(self.default_get(['move_type', 'journal_id'])))._get_edi_decoder(file_data, new=new)
```

Comment:
This will just retrieve the decoder "as if the invoice was actually created"
The invoice is created below, in `_get_edi_creation`, when called on the empty `current_invoice`